Fix `index_opts::hint` checks (#8954)
tarantool/tarantool: changelogs/unreleased/gh-8937-memtx-tree-index-doesnt-accept-true-option-for-hint.md

## bugfix/core

> * Fixed a bug when it was not possible to set the `hint` option to `true`
>   for the TREE indexes (gh-8937).

Reviewer: ```suggestion
* Fixed a bug because of which it was impossible to set the `hint` option
  to `true` for TREE indexes (gh-8937).
```

Author: Done.
